ui/app/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -0,0 +1,325 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ethUtil from 'ethereumjs-util'
+import classnames from 'classnames'
+import { ObjectInspector } from 'react-inspector'
+
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../../../../app/scripts/lib/enums'
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+import Identicon from '../../ui/identicon'
+import AccountListItem from '../../../pages/send/account-list-item/account-list-item.component'
+import { conversionUtil } from '../../../helpers/utils/conversion-util'
+import Button from '../../ui/button'
+import { DEFAULT_ROUTE } from '../../../helpers/constants/routes'
+
+export default class SignatureRequestOriginal extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  }
+
+  static propTypes = {
+    balance: PropTypes.any,
+    cancel: PropTypes.func,
+    clearConfirmTransaction: PropTypes.func,
+    conversionRate: PropTypes.any,
+    history: PropTypes.object,
+    requesterAddress: PropTypes.any,
+    selectedAccount: PropTypes.string,
+    sign: PropTypes.any,
+    txData: PropTypes.any,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      selectedAccount: props.selectedAccount,
+    }
+  }
+
+  componentDidMount = () => {
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_NOTIFICATION) {
+      window.addEventListener('beforeunload', this._beforeUnload)
+    }
+  }
+
+  componentWillUnmount = () => {
+    this._removeBeforeUnload()
+  }
+
+  _beforeUnload = (event) => {
+    const { clearConfirmTransaction, cancel } = this.props
+    const { metricsEvent } = this.context
+    metricsEvent({
+      eventOpts: {
+        category: 'Transactions',
+        action: 'Sign Request',
+        name: 'Cancel Sig Request Via Notification Close',
+      },
+    })
+    clearConfirmTransaction()
+    cancel(event)
+  }
+
+  _removeBeforeUnload = () => {
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_NOTIFICATION) {
+      window.removeEventListener('beforeunload', this._beforeUnload)
+    }
+  }
+
+  renderHeader = () => {
+    return (
+      <div className="request-signature__header">
+        <div className="request-signature__header-background" />
+
+        <div className="request-signature__header__text">
+          { this.context.t('sigRequest') }
+        </div>
+
+        <div className="request-signature__header__tip-container">
+          <div className="request-signature__header__tip" />
+        </div>
+      </div>
+    )
+  }
+
+  renderAccount = () => {
+    const { selectedAccount } = this.state
+
+    return (
+      <div className="request-signature__account">
+        <div className="request-signature__account-text">
+          { this.context.t('account') + ':' }
```

Comment:
Can we use a template string here for consistency? (A template string is used below.)